### ISSUE TITLE
Add a keymap for VS Code

### ISSUE THREAD

**Issue by anthonyvdotbe** [apache/netbeans#6458]
### Description

As the title says

### Use case/motivation

VS Code is my primary IDE, so I'd like to be able to use the same shortcuts in NetBeans.
In general I think most Java developers will start with either IntelliJ IDEA or VS Code, so if NetBeans wants to attract new users, it should offer them a smooth transition. There's already a keymap for IntelliJ IDEA, but not yet for VS Code.

### Are you willing to submit a pull request?

No

**Comment by mbien** [comment]
added https://github.com/apache/netbeans/labels/Contribution%20welcome + https://github.com/apache/netbeans/labels/good%20first%20issue labels

**Comment by vikian050194** [comment]
Hello! I'm also VS Code + NetBeans user and have some inerest in such feature so I'm working on it, unfortunately no ETA

**Comment by rothardo** [comment]
I am willing to work on this @mbien please assign this to me

**Comment by mbien** [comment]
a starting point would be to take a look at the eclipse or intellij keybindings which are here:
https://github.com/apache/netbeans/blob/master/ide/defaults/src/org/netbeans/modules/defaults/Eclipse-keybindings.xml#L24
and
https://github.com/apache/netbeans/blob/master/java/java.editor/src/org/netbeans/modules/java/editor/resources/eclipse-global-keymap.xml#L30

We don't really assign issues, you can simply start working on something and open a PR if you have any results.

**Comment by rothardo** [comment]
@mbien  I would like to inform you that i have started working on the same, I hope as I am new to open source you would be there wherever I will need help, thank you for the opportunity.

**Comment by rothardo** [comment]
I have created a keybinding xml file named `VsCode-keybindings.xml`

![keybinds](https://github.com/apache/netbeans/assets/83207402/90aeb5dd-5669-454e-b266-83f7c24857a3)

and added a profile `Keymaps/Vscode=VsCode` in `Bundle.properties`
![profilename](https://github.com/apache/netbeans/assets/83207402/4b397395-dfc1-44e0-a4e3-cff5b9a3bc8f)

an in `mf-layer.xml `file
in  `Keybindings` folder tag
I added 
```
<folder name="VsCode">
                <folder name="Defaults">
                    <file name="org-netbeans-modules-defaults-keybindings.xml" url="VsCode-keybindings.xml"/>
                </folder>
            </folder>
```
![binds](https://github.com/apache/netbeans/assets/83207402/95152870-3c48-4e37-b7f8-32d8a35f4d8f)


but I am not able to understand 
the keymaps foldering and how the ` ...  stringvalue="Actions/Tools/org-netbeans-modules-jumpto-file-FileSearchAction.instance"/>` works
I am stuck here though I have added a keymap folder named `VsCode`

![code](https://github.com/apache/netbeans/assets/83207402/d3a06fbe-515e-473d-bc38-4e660b116868)

I don't get how to proceed with this may I get some help?

**Comment by mbien** [comment]
@rothardo this looks correct to me so far - feel free to open a draft PR even if it is not finished. It is easier to comment on code there.

**Comment by rothardo** [comment]
Hi @mbien  ,

I have opened a draft PR to address the issue of migrating VSCode keybindings to NetBeans. I would greatly appreciate your assistance or a referral to a contributor who has in-depth knowledge of this area.

Thank you!

**Comment by rothardo** [comment]
Hi there, can anyone tell me how to proceed with this? this is the [Draft PR](https://github.com/apache/netbeans/pull/7682) for the same, I am close to the solution but it would be better to get guidance for someone who has previous experience in this area

**Comment by Stuaarts** [comment]
Hi, I'd like to work on this issue as my first contribution. Is it still available ? If so, Could you please assign the issue to me? Thank you !

**Comment by mbien** [comment]
@Stuaarts yes this is still open and hasn't been implemented yet. Feel free to try it and open a draft PR if you think it is working.

**Comment by king-407** [comment]
Hi, I opened a draft PR for this issue here: #9608

The PR adds a `Visual Studio Code` keymap profile using the existing NetBeans keymap infrastructure, including common editor, navigation, search, project, terminal, and debugging shortcuts.

It also adds macOS-specific keybinding overrides so the profile follows platform conventions on Mac.

Happy to adjust the shortcut set or scope based on any feedback.
